File: SDKs/cocos2dx/Classes/IntelliVerseX/IVXAINPCDialogManager.cpp
// Copyright (c) 2026 Intelli-verse-X
// MIT License — see LICENSE in the project root.

#include "IntelliVerseX/IVXAINPCDialogManager.h"
#include "cocos2d.h"
#include <stdexcept>

namespace IntelliVerseX {

IVXAINPCDialogManager& IVXAINPCDialogManager::getInstance() {
    static IVXAINPCDialogManager instance;
    return instance;
}

bool IVXAINPCDialogManager::isInitialized() const {
    return false;
}

void IVXAINPCDialogManager::initialize(void*) {
    cocos2d::log("[IVX-Cocos] IVXAINPCDialogManager::initialize: stub — not yet implemented. AI features will return empty results.");
}

void IVXAINPCDialogManager::setAuthToken(const std::string&) {
    cocos2d::log("%s", "[IVX-Cocos] AI stub: not yet implemented — stub only");
    throw std::runtime_error("Not yet implemented — stub only");
}

void IVXAINPCDialogManager::registerNPC(const IVXAINPCProfile&) {
    cocos2d::log("%s", "[IVX-Cocos] AI stub: not yet implemented — stub only");
    throw std::runtime_error("Not yet implemented — stub only");
}

void IVXAINPCDialogManager::unregisterNPC(const std::string&) {
    cocos2d::log("%s", "[IVX-Cocos] AI stub: not yet implemented — stub only");
    throw std::runtime_error("Not yet implemented — stub only");
}

void IVXAINPCDialogManager::startDialog(const std::string&, const std::string&, const std::string&,
                                        std::function<void(const IVXAINPCDialogSession&)>, ErrorCallback onError) {
    cocos2d::log("%s", "[IVX-Cocos] AI stub: not yet implemented — stub only");
    if (onError) onError({-1, "Not yet implemented — stub only"});
}

void IVXAINPCDialogManager::sendMessage(const std::string&, const std::string&,
                                        std::function<void(const std::string&)>, ErrorCallback onError) {
    cocos2d::log("%s", "[IVX-Cocos] AI stub: not yet implemented — stub only");
    if (onError) onError({-1, "Not yet implemented — stub only"});
}

void IVXAINPCDialogManager::endDialog(const std::string&, SuccessCallback, ErrorCallback onError) {
    cocos2d::log("%s", "[IVX-Cocos] AI stub: not yet implemented — stub only");
    if (onError) onError({-1, "Not yet implemented — stub only"});
}

const IVXAINPCDialogSession* IVXAINPCDialogManager::getSession(const std::string&) const {
    return nullptr;
}

std::vector<IVXAINPCDialogSession> IVXAINPCDialogManager::getSessionsForNPC(const std::string&) const {
    return {};
}

} // namespace IntelliVerseX

File: SDKs/java/src/main/java/com/intelliversex/sdk/ai/IVXAINPCDialogManager.java
// Copyright (c) 2026 Intelli-verse-X
// MIT License — see LICENSE in the project root.

package com.intelliversex.sdk.ai;

import java.util.List;
import java.util.concurrent.CompletableFuture;

/**
 * NPC profile registration and dialog sessions (Unity {@code IVXAINPCDialogManager}).
 */
public final class IVXAINPCDialogManager {

    private static final IVXAINPCDialogManager INSTANCE = new IVXAINPCDialogManager();

    private IVXAINPCDialogManager() {}

    public static IVXAINPCDialogManager getInstance() {
        return INSTANCE;
    }

    public boolean isInitialized() {
        return false;
    }

    public void initialize(Object config) {
        throw new UnsupportedOperationException("Not yet implemented — stub only");
    }

    public void setAuthToken(String token) {
        throw new UnsupportedOperationException("Not yet implemented — stub only");
    }

    public void registerNPC(IVXAINPCProfile profile) {
        throw new UnsupportedOperationException("Not yet implemented — stub only");
    }

    public void unregisterNPC(String npcId) {
        throw new UnsupportedOperationException("Not yet implemented — stub only");
    }

    public CompletableFuture<IVXAINPCDialogSession> startDialog(
            String npcId, String playerId, String playerContext) {
        return failed();
    }

    public CompletableFuture<String> sendMessage(String sessionId, String message) {
        return failed();
    }

    public CompletableFuture<Void> endDialog(String sessionId) {
        return failed();
    }

    public IVXAINPCDialogSession getSession(String sessionId) {
        throw new UnsupportedOperationException("Not yet implemented — stub only");
    }

    public List<IVXAINPCDialogSession> getSessionsForNPC(String npcId) {
        throw new UnsupportedOperationException("Not yet implemented — stub only");
    }

    private static <T> CompletableFuture<T> failed() {
        CompletableFuture<T> f = new CompletableFuture<>();
        f.completeExceptionally(new UnsupportedOperationException("Not yet implemented — stub only"));
        return f;
    }

    public static final class IVXAINPCProfile {
        public String npcId;
        public int maxTurns;
    }

    public static final class IVXAINPCDialogSession {
        public String sessionId;
        public String npcId;
        public String playerId;
    }
}
